### PR TITLE
feat(runner): make source map support optional

### LINKED ISSUE #2603: protractor registers global `source-map-support`
i found this line in the code
https://github.com/angular/protractor/blob/master/lib/runner.js#L162
and it messes up with my project's environment. I agree that some globals like `$` or `protractor` can be an exception in test environment, but things like that should be at least optional.
Is there any way it can be disabled/configured?

### PR DESCRIPTION
**Pull request by sjelin**
Closes #2603

### CODE REVIEW

File: docs/referenceConf.js [hunk 1]
```diff
@@ -311,5 +311,12 @@ exports.config = {
     tags: '@dev',
     // How to format features (default: progress)
     format: 'summary'
-  }
+  },
+
+  // See docs/plugins.md
+  plugins: [],
+
+  // Turn of source map support.  Stops protractor from registering global
```
Comment (hankduan) [inline docs/referenceConf.js]: off

**Comment by hankduan** [comment]
Optimist doesn't handle string parsing. So `protractor conf.js --sourceMap false` will be read as 'sourceMap="false"', which will evaluate to truthy. 

Users then have to do 'protractor conf.js --sourceMap 0' or 'protractor conf.js --no-sourceMap', which they will probably not know and feel it's a bug. 

Let's call it --skipSourceMapSupport, and flip the boolean.

**Comment by sjelin** [comment]
Comments addressed

File: docs/referenceConf.js [hunk 2]
```diff
@@ -311,5 +311,12 @@ exports.config = {
     tags: '@dev',
     // How to format features (default: progress)
     format: 'summary'
-  }
+  },
+
+  // See docs/plugins.md
+  plugins: [],
+
+  // Turn off source map support.  Stops protractor from registering global
```
Comment (hankduan) [inline docs/referenceConf.js]: nit: Turns off

**Comment by sjelin** [comment]
Merged in with 4924028

**Pull request closed without merging**